Small fixes for building with Nightly compiler

b-zee opened the pull request:
- fix: update curve25519-dalek dep
- fix: apply suspicious_open_options from clippy

## Description

reviewpad:summary

davidrusu commented on sn_node/src/comm/peer_session.rs:
```diff
@@ -184,7 +188,12 @@ impl PeerSessionWorker {
+    async fn send(&mut self, mut job: SendJob) -> SessionStatus {
+        if job.retries > MAX_SENDJOB_RETRIES {
+            job.reporter.send(SendStatus::MaxRetriesReached);
+            return SessionStatus::Ok;
```
Debatable whether we should be terminating this peer connection or if we should just continue.

On one hand the errors may be specific to the SendJob, i.e. suppose the message is just too big to stuff into the pipe.

On the other, we gave this peer connection 10 attempts to complete the job so perhaps it's worth terminating and having higher level code try and re-create the session.

Thoughts?

joshuef replied:
I think a terminate can make sense here. 10 attempts is a good amount. This _shouldnt_ be happening all the time, so it seems fine to me.

b-zee committed:
fix: update curve25519-dalek dep

This makes Rust nightly compiler happy again:
https://github.com/dalek-cryptography/curve25519-dalek/pull/619

b-zee committed:
fix: apply suspicious_open_options from clippy
